src/common.rs

```diff
@@ -1276,10 +1284,188 @@ fn path_directories() -> Vec<std::path::PathBuf> {
     dirs
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ValidatorsTable {
+    pub validator_id: near_primitives::types::AccountId,
+    pub fee: RewardFeeFraction,
+    pub delegators: u64,
+    pub stake: near_primitives::types::Balance,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Deserialize)]
+pub struct RewardFeeFraction {
+    pub numerator: u32,
+    pub denominator: u32,
+}
+
+pub fn get_validator_list(
+    network_config: &crate::config::NetworkConfig,
+) -> color_eyre::eyre::Result<Vec<ValidatorsTable>> {
+    let json_rpc_client = network_config.json_rpc_client();
+
+    let validators_stake = get_validators_stake(&json_rpc_client)?;
+
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()?;
+    let chunk_size = 15;
+    let concurrency = 10;
+
+    let validators_table: std::collections::HashMap<
+        near_primitives::types::AccountId,
+        ValidatorsTable,
+    > = runtime
+        .block_on(
+            futures::stream::iter(
+                validators_stake
+                    .iter()
+                    .collect::<Vec<_>>()
+                    .chunks(chunk_size),
+            )
+            .map(|validators| async { get_validators_table(&json_rpc_client, validators).await })
+            .buffer_unordered(concurrency)
+            .collect::<Vec<Result<_, _>>>(),
+        )
+        .into_iter()
+        .try_fold(std::collections::HashMap::new(), |mut acc, x| {
+            acc.extend(x?);
+            Ok::<_, color_eyre::eyre::Error>(acc)
+        })?;
+
+    let mut validator_list: Vec<ValidatorsTable> = validators_table.into_values().collect();
+    validator_list.sort_by(|a, b| b.stake.cmp(&a.stake));
+    Ok(validator_list)
```

Comment:
1. Follow the refactoring done in https://github.com/near/near-cli-rs/pull/227/commits/dbf79fc9bfabea20262f51c9337b84734582ee02
2. Collect directly to Vec as you don't need HashMap.

Reply:
Done